### DISCUSSION TITLE
post/advent-of-code-2022-lessons-learned

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#3] [Announcements]
# post/advent-of-code-2022-lessons-learned

Mark Heath's Development Blog

https://markheath.net/post/advent-of-code-2022-lessons-learned

**Comment by Octanoz** [comment]
Hi Mark,
I started coding earlier this year (C# almost exclusively so far) and was very happy to find out about Advent of Code. Like you said, it's great how the second part can mean just a little tweak to your existing method or an entirely different approach!

I didn't make it past day 15 or 16 of 2022 but I definitely wanted more so I started working on the 2015 challenges. I managed to get almost all of them done by myself - there was something about MD5 hashing which probably made more sense to people during the crypto craze... Part of my success can probably also be blamed on the better hardware that I have at my disposal compared to people that would have tried them in 2015.

I do like to check how other people completed them and see if there's any clever things I missed that I can use myself. Most of the Python code out there is written by competitive coders that like to write their code like a mathematician would write their formulas so not really helpful for me.

That's when I stumbled upon your channel and when I first checked out the videos I found myself actually averse to your solutions because they were exclusively using LINQ. A few days later I realized that I had been conditioned by LeetCode that practically forces you to code every line yourself instead of really playing to your language's strengths. And I had really 
been enjoying LINQ before!

So thanks for opening my eyes! I'm enjoying trying out the Morelinq methods right now. I just read that I need to check out Superlinq as well and I'll have a go at your LINQ course in the coming weeks.

Hope to see some Advent of Code 2023 videos later this year! Good luck anyways! ;)

**Reply by markheath** [reply]
hi Octanoz, great to hear from you and glad you enjoyed the videos. Don't know if I'll have time for Advent of Code this year, but have enjoyed doing it with my son the last two years who has been learning C# at college.
